app/src/main/java/com/kotlin/readyplayerme/UrlBuilder.kt

```diff
@@ -0,0 +1,82 @@
+package com.kotlin.readyplayerme
+
+
+data class UrlConfig(
+    var subdomain: String = "demo",
+    var clearCache: Boolean = false,
+    var quickStart: Boolean = false,
+    var gender: Gender = Gender.NONE,
+    var bodyType: BodyType = BodyType.SELECTABLE,
+    var loginToken: String = "",
+    var language: Language = Language.DEFAULT
+)
+
+class UrlBuilder(
+    private val urlConfig: UrlConfig = UrlConfig()
+) {
+    companion object {
+        private const val CLEAR_CACHE_PARAM = "clearCache"
+        private const val FRAME_API_PARAM = "frameApi"
+        private const val QUICK_START_PARAM = "quickStart"
+        private const val SELECT_BODY_PARAM = "selectBodyType"
+        private const val LOGIN_TOKEN_PARAM = "token"
+    }
+
+    fun buildUrl(): String {
+        val baseUrl = "https://${urlConfig.subdomain}.readyplayer.me/"
+        val builder = StringBuilder(baseUrl)
+
+        if (urlConfig.language != Language.DEFAULT) {
+            builder.append("${urlConfig.language}/")
+        }
+
+        builder.append("avatar?$FRAME_API_PARAM")
+
+        if (urlConfig.clearCache) {
+            builder.append("&$CLEAR_CACHE_PARAM")
+        }
+
+        if (urlConfig.loginToken.isNotEmpty()) {
+            builder.append("&$LOGIN_TOKEN_PARAM=$urlConfig.loginToken")
+        }
+
+        if (urlConfig.quickStart) {
+            builder.append("&$QUICK_START_PARAM")
+        } else {
+            if (urlConfig.gender != Gender.NONE) {
+                builder.append("&gender=${urlConfig.gender}")
+            }
+            builder.append(if (urlConfig.bodyType == BodyType.SELECTABLE) "&$SELECT_BODY_PARAM" else "&bodyType=${urlConfig.bodyType}")
```

Comment:
I took the approach of flattening this out a bit inside the iOS example, as having multiple ifs nested inside the else of another if is not the easiest code to follow, and isn't a great standard to set for an open-source example. Perhaps it might be worth thinking about the same here?